### PR TITLE
expose pulpcore's HIDE_GUARDED_DISTRIBUTIONS setting

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

**Comment by evgeni** [comment]
requires https://github.com/theforeman/puppet-pulpcore/pull/292

File: manifests/init.pp [hunk 1]
```diff
@@ -144,6 +146,7 @@
   Variant[Stdlib::Absolutepath, Array[Stdlib::Absolutepath]] $pulpcore_additional_import_paths = [],
   Variant[Stdlib::Absolutepath, Array[Stdlib::Absolutepath]] $pulpcore_additional_export_paths = [],
   Boolean $pulpcore_telemetry = false,
+  Boolean $pulpcore_hide_guarded_distributions = false,
```
Comment (evgeni) [inline manifests/init.pp:149]:
```suggestion
  Optional[Boolean] $pulpcore_hide_guarded_distributions = undef,
```

I'm torn :)

Comment (evgeni) [reply manifests/init.pp:149]: or `= true`?

Comment (evgeni) [reply manifests/init.pp:149]: Both, Jameer and I think it should be `true` in the opinionated Katello deployment.

Comment (ekohl) [reply manifests/init.pp:149]: I'm also leaning to `true`.

Comment (evgeni) [reply manifests/init.pp]: it has been `true`d

File: metadata.json [hunk 2]
```diff
@@ -38,7 +38,7 @@
     },
     {
       "name": "theforeman/pulpcore",
-      "version_requirement": ">= 7.1.0 < 9.0.0"
+      "version_requirement": ">= 8.2.0 < 9.0.0"
```
[the remark this answers is not in the document]
Comment (ekohl) [reply metadata.json:41]: Given https://github.com/theforeman/puppet-pulpcore/pull/293 I think it's a fairly safe assumption.

**Pull request merged**